sys/net/gnrc/network_layer/sixlowpan/frag/sfr/gnrc_sixlowpan_frag_sfr.c

```diff
@@ -0,0 +1,1655 @@
+
+            if (last_sent_since <= CONFIG_GNRC_SIXLOWPAN_SFR_INTER_FRAME_GAP_US) {
+                uint32_t offset = CONFIG_GNRC_SIXLOWPAN_SFR_INTER_FRAME_GAP_US -
+                                  last_sent_since;
+                DEBUG("6lo sfr: arming inter-frame timer in %" PRIu32 " us\n",
+                      last_sent_since);
+                xtimer_set_msg(&_if_gap_timer, offset, &_if_gap_msg, _getpid());
+            }
+            else {
+                DEBUG("6lo sfr: send frame immediately\n");
+                gnrc_sixlowpan_frag_sfr_inter_frame_gap();
+            }
+        }
+        else {
+            DEBUG("6lo sfr: inter-frame timer was already set\n");
+        }
+    }
+}
+
+static void _sched_arq_timeout(gnrc_sixlowpan_frag_fb_t *fbuf, uint32_t offset)
+{
+    if (fbuf->sfr.arq_timeout_event.msg.content.ptr != NULL) {
+        DEBUG("6lo sfr: ARQ timeout for datagram %u already scheduled\n",
+              (uint8_t)fbuf->tag);
+        return;
+    }
+    DEBUG("6lo sfr: arming ACK timeout in %lums for datagram %u\n",
+          (long unsigned)offset, fbuf->tag);
+    fbuf->sfr.arq_timeout_event.event.offset = offset;
+    fbuf->sfr.arq_timeout_event.msg.content.ptr = fbuf;
+    fbuf->sfr.arq_timeout_event.msg.type = GNRC_SIXLOWPAN_FRAG_SFR_ARQ_TIMEOUT_MSG;
+    evtimer_add_msg(&_arq_timer, &fbuf->sfr.arq_timeout_event,
+                    _getpid());
+}
+
+static void _sched_abort_timeout(gnrc_sixlowpan_frag_fb_t *fbuf)
+{
+    /* no fragments to wait for anymore as we aborted fragmentation and just
+     * wait for an ACK by the reassembling end point that they know. As such,
+     * clean-out the fragmentation buffer. */
+    _clean_slate_datagram(fbuf);
+    fbuf->sfr.retrans = 0;
+    _sched_arq_timeout(fbuf, fbuf->sfr.arq_timeout);
+}
+
+static void _handle_rfrag(gnrc_netif_hdr_t *netif_hdr, gnrc_pktsnip_t *pkt,
+                          unsigned page)
+{
+    _generic_rb_entry_t entry = { .type = _UNDEF };
+
+    (void)page;
+    if (sixlowpan_sfr_rfrag_get_seq(pkt->data) == 0U) {
+        _handle_1st_rfrag(netif_hdr, pkt, page, &entry);
+    }
+    else {
+        _handle_nth_rfrag(netif_hdr, pkt, page, &entry);
+    }
+}
+
+static void _handle_ack(gnrc_netif_hdr_t *netif_hdr, gnrc_pktsnip_t *pkt,
+                        unsigned page)
+{
+    gnrc_sixlowpan_frag_vrb_t *vrbe;
+    sixlowpan_sfr_ack_t *hdr = pkt->data;
+
+    (void)page;
+    DEBUG("6lo sfr: received ACK for datagram (%s, %02x): %02X%02X%02X%02X\n",
+          gnrc_netif_addr_to_str(gnrc_netif_hdr_get_src_addr(netif_hdr),
+                                 netif_hdr->src_l2addr_len,
+                                 addr_str), hdr->base.tag,
+          hdr->bitmap[0], hdr->bitmap[1], hdr->bitmap[2], hdr->bitmap[3]);
+    if ((vrbe = gnrc_sixlowpan_frag_vrb_reverse(
+            gnrc_netif_hdr_get_netif(netif_hdr),
+            gnrc_netif_hdr_get_src_addr(netif_hdr),
+            netif_hdr->src_l2addr_len, hdr->base.tag)) != NULL) {
```

Comment:
What case is this?

Reply:
I also use DEBUG messages sometimes to comment on stuff... Should I really repeat what is stated there?